homeassistant/components/yamaha/media_player.py

```diff
@@ -205,24 +228,38 @@ def __init__(self, name, receiver, source_ignore, source_names, zone_names):
         self._is_playback_supported = False
         self._play_status = None
         self._name = name
-        self._zone = receiver.zone
-        if self.receiver.serial_number is not None:
+        self._zone = zctrl.zone
+        if self.zctrl.serial_number is not None:
             # Since not all receivers will have a serial number and set a unique id
             # the default name of the integration may not be changed
             # to avoid a breaking change.
-            self._attr_unique_id = f"{self.receiver.serial_number}_{self._zone}"
+            # Prefix as MusicCast could have used this
+            self._attr_unique_id = f"{DOMAIN}_{self.zctrl.serial_number}_{self._zone}"
```

Comment:
If I understand correctly, this changes the unique_id of the entities, we should not change that without a migration. For me its unclear what this is actually achieving right now since the unique id is the same but with the domain added in front

Reply:
I agree and for Yamaha MusicCast enabled amps that the right solution, I was worried about this if I fixed up auto discovery but given I have the last generation of amp without it and that's 9 years old, I think the code will be good enough as is without this discovery, or at last deal with this then...

Reply:
Hi, def _discovery (retrieve more info(serial)) if we can via discovery even for static host entries  now and def async_setup_platform is changed to cope with the receiver not being there on planform init.